{
  final Errors errors=new Errors();
  try {
    T t=injector.callInContext(new ContextualCallable<T>(){
      @Override public T call(      InternalContext context) throws ErrorsException {
        Dependency dependency=context.getDependency();
        return internalFactory.get(errors,context,dependency);
      }
    }
);
    errors.throwIfNewErrors(0);
    return t;
  }
 catch (  ErrorsException e) {
    throw new ProvisionException(errors.merge(e.getErrors()).getMessages());
  }
}
